frontend/src/common/PageWithTabs.tsx

```diff
@@ -35,10 +37,14 @@ export default function PageWithTabs({
   }, [tab, setCurrentTab, tabs])
 
   const handleChange = (_event: SyntheticEvent, newValue: string) => {
-    setCurrentTab(newValue)
-    router.replace({
-      query: { ...router.query, tab: newValue },
-    })
+    if (unsavedChanges) {
+      sendWarning()
+    } else {
+      setCurrentTab(newValue)
```

Comment:
If the user confirms that they do want to change tabs does this logic get hit? It looks like it wouldn't to me.

Maybe we need `sendWarning` to return the user's response to the popup window? We can then conditionally continue with the next piece of logic as normal, if the user confirmed, or skip the logic, if the user canceled.